### PR TITLE
feat: Update XDS icon classes

### PR DESCRIPTION
**Pull request by annacv**
[EX-7776](https://searchbroker.atlassian.net/browse/EX-7776)

* Renamed some icons to follow the naming convention `x-icon-${iconName}`
* Updated size classes
* Fix clear input btn in mobile to set it squared

[EX-7776]: https://searchbroker.atlassian.net/browse/EX-7776?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/design-system/tokens.scss [hunk 1]
```diff
@@ -1,6 +1,6 @@
 :root {
   // Icons
```
Comment (tiborux) [inline src/design-system/tokens.scss:2]: is this needed?

**Comment by tiborux** [comment]
Update `x-components` to the latest version, if not, you are going to mix the old styles with the new ones.

File: src/components/column-picker.vue [hunk 2]
```diff
@@ -10,8 +10,8 @@
     >
       <component
         :is="icon(column)"
-        class="x-icon--l"
-        :class="{ 'x-text-neutral-90': isSelected }"
+        class="x-icon-lg x-icon-stroke-width-lg"
```
Comment (tiborux) [inline src/components/column-picker.vue:13]: Its not needed

```suggestion
        class="x-icon-lg"
```

File: src/components/search/facets/custom-facets.vue [hunk 3]
```diff
@@ -11,17 +11,17 @@
         </template>
         <template #default>
           <AllFilter v-slot="{ isSelected }" :facet="facet">
-            <CheckboxCardSelectedIcon v-if="isSelected" class="x-icon--l" />
-            <CheckboxCardUnselectedIcon v-else class="x-icon--l" />
+            <CheckboxCardSelectedIcon v-if="isSelected" class="x-icon-lg" />
```
Comment (tiborux) [inline src/components/search/facets/custom-facets.vue:14]: After updating `x-components` you will need to change these components by:

CheckboxSelectedIcon
CheckboxUnselectedIcon

File: src/components/my-history/custom-my-history.vue [hunk 4]
```diff
@@ -83,7 +83,9 @@
                   <div
                     class="x-list x-list--horizontal x-list--gap-05 x-padding--bottom-02 x-padding--top-02"
                   >
-                    <HistoryIcon :class="{ 'x-icon--l': $x.device === 'mobile' }" />
+                    <HistoryIcon
+                      :class="{ 'x-icon-lg x-icon-stroke-width-lg': $x.device === 'mobile' }"
```
Comment (tiborux) [inline src/components/my-history/custom-my-history.vue:87]: its not needed
```suggestion
                      :class="{ 'x-icon-lg': $x.device === 'mobile' }"
```

File: src/components/my-history/custom-my-history.vue [hunk 5]
```diff
@@ -101,7 +103,7 @@
                   </div>
 
                   <template #remove-button-content>
-                    <CrossTinyIcon :class="{ 'x-icon--l': $x.device === 'mobile' }" />
+                    <CrossTinyIcon :class="{ 'x-icon-lg': $x.device === 'mobile' }" />
```
Comment (tiborux) [inline src/components/my-history/custom-my-history.vue:106]: After talking with @cris-crash, change the component to be lg always:

```suggestion
                    <CrossTinyIcon class="x-icon-lg" />
```

**Comment by tiborux** [comment]
Change the `filtersIcon` to use lg size:

`<FiltersIcon class="x-icon-lg" />` in `desktop-toolbar` and `mobile-open-aside`

File: src/components/predictive-layer/predictive-layer.vue [hunk 6]
```diff
@@ -140,7 +140,7 @@
                 />
                 <LightBulbOn
                   v-else
-                  class="x-icon--light-bulb-on"
+                  class="x-icon-light-bulb-on"
```
Comment (mnavarroespinosa) [inline src/components/predictive-layer/predictive-layer.vue:143]: This class can be removed and the style applied to it in `tokens.scss`

File: src/design-system/tokens.scss [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
-  .x-icon--light-bulb-on {
+  .x-icon-light-bulb-on {
     --x-size-stroke-width-icon-default: 0.8px;
   }
```
Comment (mnavarroespinosa) [inline src/design-system/tokens.scss:5]: This can be removed as this token does not exist in the new XDS icon.

**Review by mnavarroespinosa — CHANGES_REQUESTED** [review]
The `x-components` version, in which the old icon DS has been deprecated, needs to be updated in the `package.json`, if I'm not wrong it is the `^3.0.0-alpha.308` version.

File: src/design-system/tokens.scss (hunk 1 above)
Comment (annacv) [reply src/design-system/tokens.scss:2]: I see @mnavarroespinosa comment and I agree then, we shouldn't use values that are not expected in XDS in the archetype. So I will remove this. Thks!

**Review by mnavarroespinosa — DISMISSED** [review]
LGTM, good job!

**Review by tiborux — APPROVED** [review]

**Review by mnavarroespinosa — APPROVED** [review]

**Pull request merged**